src/screens/UserPortal/Campaigns/Campaigns.tsx

```
@@ -95,96 +78,65 @@ const Campaigns = (): JSX.Element => {
-  // Redirects to the homepage if orgId or userId is missing
   if (!orgId || !userId) {
     return <Navigate to={'/'} replace />;
   }
 
-  /**
-   * Opens the modal for adding a pledge to a selected campaign.
-   *
-   * @param campaign - The campaign to which the user wants to add a pledge.
-   */
-  const openModal = (campaign: InterfaceUserCampaign): void => {
+  const openModal = useCallback((campaign: InterfaceUserCampaign): void => {
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

Hooks should not be called after an early return.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)